lib/modules/datasource/packagist/index.ts

```diff
@@ -115,17 +115,28 @@ export class PackagistDatasource extends Datasource {
 
   @cache({
     namespace: `datasource-${PackagistDatasource.id}-org`,
-    key: (regUrl: string) => regUrl,
+    key: (registryUrl: string, metadataUrl: string, packageName: string) =>
+      `${registryUrl}:${metadataUrl}:${packageName}`,
     ttlMinutes: 10,
   })
-  async packagistOrgLookup(name: string): Promise<ReleaseResult | null> {
-    const regUrl = 'https://packagist.org';
-    const pkgUrl = joinUrlParts(regUrl, `/p2/${name}.json`);
-    const devUrl = joinUrlParts(regUrl, `/p2/${name}~dev.json`);
-    const results = await p.map([pkgUrl, devUrl], (url) =>
-      this.http.getJson(url).then(({ body }) => body)
+  async packagistV2Lookup(
+    registryUrl: string,
+    metadataUrl: string,
+    packageName: string
+  ): Promise<ReleaseResult | null> {
+    let pkgUrl = metadataUrl.replace('%package%', packageName);
+    pkgUrl = parseUrl(pkgUrl) ? pkgUrl : resolveBaseUrl(registryUrl, pkgUrl);
+    const pkgPromise = this.getJson(pkgUrl, z.unknown());
```

Comment:
the `unknown` is temporary? can we just remove it for now? or is the parameter required?

Reply:
It's being parsed a bit later, once all requests are performed.
I want to make the Packagist datasource completely schema-driven, but it requires more refactoring PRs in the future